src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceCreationService.kt

```diff
@@ -0,0 +1,228 @@
+      "No active offender manager found for $nomsId",
+    )
+    val currentResponsibleOfficerDetails = offenderManagers.find {
+      it.staffCode == currentActiveOffenderManager.staffDetail.code
+    } ?: error("No responsible officer details found for $nomsId")
+
+    val responsibleCom = staffRepository.findByStaffIdentifier(currentResponsibleOfficerDetails.staffId)
+      ?: error("Staff with staffIdentifier ${currentResponsibleOfficerDetails.staffId} not found")
+
+    val createdBy = staffRepository.findByUsernameIgnoreCase(username)
+      ?: error("Staff with username $username not found")
+
+    val licenceType = getLicenceType(nomisRecord)
+
+    val createLicenceRequest = createLicenceRequest(
+      licenceType,
+      nomsId,
+      nomisRecord,
+      prisonInformation,
+      currentResponsibleOfficerDetails,
+      deliusRecord,
+    )
+
+    val licenceEqualityCheck = checkCreatedLicence(request, createLicenceRequest)
+      .filter { (test, _) -> !test }
+      .map { (_, field) -> field }
+
+    val licence = transform(createLicenceRequest)
+
+    licence.dateCreated = LocalDateTime.now()
+    licence.responsibleCom = responsibleCom
+    licence.createdBy = createdBy
+    licence.updatedByUsername = username
+
+    val licenceEntity = licenceRepository.saveAndFlush(licence)
+    val createLicenceResponse = transformToLicenceSummary(licenceEntity)
+
+    val entityStandardLicenceConditions =
+      createLicenceRequest.standardLicenceConditions.transformToEntityStandard(licenceEntity, "AP")
+    val entityStandardPssConditions = createLicenceRequest.standardPssConditions.transformToEntityStandard(licenceEntity, "PSS")
+    standardConditionRepository.saveAllAndFlush(entityStandardLicenceConditions + entityStandardPssConditions)
+
+    auditEventRepository.saveAndFlush(
+      AuditEvent(
+        licenceId = createLicenceResponse.licenceId,
+        username = username,
+        fullName = "${createdBy.firstName} ${createdBy.lastName}",
+        summary = "Licence created for ${createLicenceRequest.forename} ${createLicenceRequest.surname}",
+        detail = "ID ${licenceEntity.id} type ${licenceEntity.typeCode} status ${licenceEntity.statusCode.name} version ${licenceEntity.version}",
+      ),
+    )
+
+    licenceEventRepository.saveAndFlush(
+      EntityLicenceEvent(
+        licenceId = createLicenceResponse.licenceId,
+        eventType = LicenceEventType.CREATED,
+        username = username,
+        forenames = createdBy.firstName,
+        surname = createdBy.lastName,
+        eventDescription = "Licence created for ${licenceEntity.forename} ${licenceEntity.surname}",
+      ),
+    )
+
+    if (licenceEqualityCheck.isNotEmpty()) {
+      log.warn("Licence inconsistency for ${licenceEntity.id} with the following differences: $licenceEqualityCheck")
+    } else {
+      log.info("Licence request and creation are identical")
+    }
+
+    return createLicenceResponse
+  }
+
+  private fun createLicenceRequest(
```

Comment:
obviously we can merge the request and transforming logic once we are happy this works